test/k8sT/KafkaPolicies.go

```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
 	})
 
 	AfterEach(func() {
-		By("Deleting demo path")
-		kubectl.Delete(demoPath)
+		// On aftereach don't make assertions to delete all.
+		_ = kubectl.Delete(demoPath)
+		_ = kubectl.Delete(l7Policy)
+
 		err := kubectl.WaitCleanAllTerminatingPods()
 		Expect(err).To(BeNil(), "Terminating containers are not deleted after timeout")
 
 	})
 
 	It("KafkaPolicies", func() {
-		clusterIP, err := kubectl.Get(helpers.DefaultNamespace, "svc").Filter(
-			"{.items[?(@.metadata.name == \"kafka-service\")].spec.clusterIP}")
-		Expect(err).Should(BeNil(), "Failed to get clusterIP")
-		logger.Infof("KafkaPolicyRulesTest: cluster service ip '%s'", clusterIP)
-
-		By("Waiting for all Cilium Pods and endpoints to be ready ")
-		By("Waiting for node K8s1")
-		ciliumPod1, _ := kubectl.WaitCiliumEndpointReady(podFilter, helpers.K8s1)
-
-		By("Waiting for node K8s2")
-		kubectl.WaitCiliumEndpointReady(podFilter, helpers.K8s2)
 
-		appPods := helpers.GetAppPods(apps, helpers.DefaultNamespace, kubectl, "app")
 		By("Testing basic Kafka Produce and Consume")
-
 		// We need to produce first, since consumer script waits for
 		// some messages to be already there by the producer.
-		err = kubectl.ExecKafkaPodCmd(
-			helpers.DefaultNamespace, appPods[empireHqApp], fmt.Sprintf(prodHqAnnounce))
-		Expect(err).Should(BeNil(), "Failed to produce to empire-hq on topic empire-announce")
+
+		err := kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[empireHqApp], prodHqAnnounce)
+		Expect(err).Should(BeNil(), "EmpireHQApp cannot announce HQ")
 
 		err = kubectl.ExecKafkaPodCmd(
-			helpers.DefaultNamespace, appPods[outpostApp], fmt.Sprintf(conOutpostAnnoune))
-		Expect(err).Should(BeNil(), "Failed to consume from outpost on topic empire-announce")
+			helpers.DefaultNamespace, appPods[outpostApp], conOutpostAnnoune)
+		Expect(err).Should(BeNil(), "OutpostApp cannot announce Outpost")
 
 		err = kubectl.ExecKafkaPodCmd(
-			helpers.DefaultNamespace, appPods[empireHqApp], fmt.Sprintf(prodHqDeathStar))
-		Expect(err).Should(BeNil(), "Failed to produce to empire-hq on topic deathstar-plans")
+			helpers.DefaultNamespace, appPods[empireHqApp], prodHqDeathStar)
+		Expect(err).Should(BeNil(), "EmpireHqApp cannot produce messages")
 
 		err = kubectl.ExecKafkaPodCmd(
-			helpers.DefaultNamespace, appPods[outpostApp], fmt.Sprintf(conOutDeathStar))
-		Expect(err).Should(BeNil(), "Failed to consume from outpost on topic deathstar-plans")
+			helpers.DefaultNamespace, appPods[outpostApp], conOutDeathStar)
+		Expect(err).Should(BeNil(), "OutpostApp cannot produce messages")
 
 		err = kubectl.ExecKafkaPodCmd(
-			helpers.DefaultNamespace, appPods[backupApp], fmt.Sprintf(prodBackAnnounce))
-		Expect(err).Should(BeNil(), "Failed to produce to backup on topic empire-announce")
+			helpers.DefaultNamespace, appPods[backupApp], prodBackAnnounce)
+		Expect(err).Should(BeNil(), "BackupApp cannot not announce")
```

Comment:
`BackupApp cannot not announce` ---> `Failed to produce to backup on topic empire-announce`
This specifies exactly what action on what topic failed for the particular apps.